doc/etcd-defrag-cronjob.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: etcd-defrag
+  namespace: kube-system
+spec:
+  schedule: "14 9 * * 1-5"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+          - name: etcd-defrag
+            image: ghcr.io/ahrtr/etcd-defrag:v0.4.0
+            args:
+            - --cacert=/etc/kubernetes/pki/etcd/ca.crt
+            - --cert=/etc/kubernetes/pki/etcd/peer.crt
+            - --key=/etc/kubernetes/pki/etcd/peer.key
+            - --cluster
+            - --defrag-rule
+            - "dbQuotaUsage > 0.8 || dbSize - dbSizeInUse > 200*1024*1024"
+            volumeMounts:
+            - mountPath: /etc/kubernetes/pki/etcd
+              name: etcd-certs
+          volumes:
+          - hostPath:
+              path: /etc/kubernetes/pki/etcd
+              type: Directory
+            name: etcd-certs
+          restartPolicy: OnFailure
+          hostNetwork: true
+          securityContext:
+            runAsUser: 0
+          tolerations:
+          - effect: NoSchedule
+            key: node-role.kubernetes.io/control-plane
+          - effect: NoExecute
+            key: node.kubernetes.io/not-ready
+            operator: Exists
+            tolerationSeconds: 300
```

Comment:
Usually defragment is a time-consuming task. Suggest to set the default value to a bigger value, e.g. 900 (15m). Same as below

Reply:
This toleration is a [default Kubernetes toleration:](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions)
<img width="1006" alt="image" src="https://github.com/ahrtr/etcd-defrag/assets/2116997/8700dec1-a2ea-4aba-8555-f696f3dd18e6">

Unless `etcd-defrag` actually makes the node `NotReady` (which it shouldn't, in the worst case it would only impact `etcd` and/or the apiserver), the node won't be in the `NotReady` state.

I'm not sure it's worth changing that value.

The reason I manually added the 2 `not-ready` and `not-reachable` tolerations is that once you specify a toleration (the `control-plane` one here), the default tolerations are not anymore automatically added. This means that if by mishap the executing node were to be `NotReady`, the pod would instantaneously get killed.

what do you think?

Reply:
Actually the best approach would to follow exactly the same configuration as the etcd [static?] pod. If etcd tolerates NoExecute only 300s, then it doesn't make sense to configure the tool as 900s.

Reply:
that's a good point. `etcd` has the following tolerations per default:
```yaml
  tolerations:
  - effect: NoExecute
    operator: Exists
```
I will use the same, it makes sense to use the same tolerations as the `kubeadm`-generated etcd static pods.